src/zns.ts

```diff
@@ -86,9 +84,11 @@ export default class ZNS extends NamingService {
     if (!this.url) {
       throw new Error('Unspecified url in Namicorn ZNS configuration');
     }
-    this.registryAddress = RegistryMap[this.network];
+    this.registryAddress = source.registry ? source.registry :  RegistryMap[this.network];
     if (this.registryAddress)
       this.registry = this.zilliqa.contracts.at(this.registryAddress);
+    if (!this.registry)
+      throw new Error(`No registry for address: ${this.registryAddress} `);
```

Comment:
Do we need this throw? 
In theory if we don't have a registry then nothing will work here

Reply:
And if we do need it, I need to add the same for ens.ts